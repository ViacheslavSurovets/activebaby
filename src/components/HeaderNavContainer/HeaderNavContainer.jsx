import React, { useMemo } from 'react';
import {
  Dropdown,
  DropdownGroup, DropdownGroupArticle,
  DropdownGroupWrapper, DropdownLink,
  HeaderNav,
  HeaderNavLink,
  List,
  ListItem, LogoDesktopWrapper, SearchItemDesktop, HeaderNavWrapper
} from '@components/Header/styles';
import { LogoDesktop } from '@components/Logo';
import { SearchDesktop } from '@components/SearchDesktop';
import { useTranslation } from 'react-i18next';


const HeaderNavContainer = () => {

  const { t, i18n } = useTranslation ();

  const headerCategory = useMemo ( () => ([
    {
      id: 'Прогулки',
      listItems: [
        {
          text: t ( 'header.headerCategory.columnOne.strollers' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnOne.quatro' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnOne.lorelli' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnOne.carello' ),
          path: '#'
        },
      ]
    },
    {
      id: 'Модульные',
      listItems: [
        {
          text: t ( 'header.headerCategory.columnTwo.modules' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnTwo.adamex' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnTwo.used' ),
          path: '#'
        }
      ]
    },
    {
      id: 'Автолюльки',
      listItems: [
        {
          text: t ( 'header.headerCategory.columnThree.carSeats' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnThree.carlo' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnThree.adamex' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnThree.used' ),
          path: '#'
        },
      ]
    },
    {
      id: 'Аксесуары',
      listItems: [
        {
          text: t ( 'header.headerCategory.columnFour.accessories' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnFour.adapters' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnFour.tires' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnFour.cameras' ),
          path: '#'
        }
      ],
    },
    {
      id: 'Постельное белье',
      listItems: [
        {
          text: t ( 'header.headerCategory.columnFive.linens' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnFive.individualTailoring' ),
          path: '#'
        }
      ]
    }
  ]), [ i18n.language ] );
  const headerArticles = useMemo ( () => ([
    {
      text: t ( 'header.headerArticle.articleOne.title' ),
      path: '#'
    },
    {
      text: t ( 'header.headerArticle.articleTwo.title' ),
      path: '#'
    },
    {
      text: t ( 'header.headerArticle.articleThree.title' ),
      path: '#'
    }

  ]), [ i18n.language ] );


  return (
    <HeaderNav>
      <List>
        <ListItem><HeaderNavLink to="/">{ t ( 'header.links.main' ) }</HeaderNavLink></ListItem>
        <ListItem><HeaderNavLink to="#">{ t ( 'header.links.categories' ) }</HeaderNavLink>
          <Dropdown>
            <DropdownGroupWrapper>
              { headerCategory.map ( item =>
                <DropdownGroup key={ item.id }>
                  { item.listItems.map ( listItem =>
                    <ListItem key={ listItem.text }><DropdownLink
                      to={ listItem.path }>{ listItem.text }</DropdownLink></ListItem>
                  ) }
                </DropdownGroup>
              ) }
            </DropdownGroupWrapper>
          </Dropdown>
        </ListItem>
        <ListItem><HeaderNavLink to="#">{ t ( 'header.links.aboutUs' ) }</HeaderNavLink></ListItem>
      </List>

      <LogoDesktopWrapper>
        <LogoDesktop />
      </LogoDesktopWrapper>

      <List>
        <ListItem><HeaderNavLink to="/">{ t ( 'header.links.payments' ) }</HeaderNavLink></ListItem>
        <ListItem><HeaderNavLink to="#">{ t ( 'header.links.articles' ) }</HeaderNavLink>
          <Dropdown>
            <DropdownGroupWrapper>
              { headerArticles.map ( item =>
                <DropdownGroupArticle key={ item.text }>
                  <ListItem key={ item.text }><DropdownLink
                    to={ item.path }>{ item.text }</DropdownLink></ListItem>
                </DropdownGroupArticle>
              ) }
            </DropdownGroupWrapper>
          </Dropdown>
        </ListItem>

        { /* desktop/searchMediaQueryLow*/ }
        <SearchItemDesktop>
          <SearchDesktop />
        </SearchItemDesktop>

      </List>
    </HeaderNav>

  );
};

export default HeaderNavContainer;
