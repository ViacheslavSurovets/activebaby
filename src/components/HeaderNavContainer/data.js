
export const headerCategoryData = [
  {
    id: 'Прогулки',
    listItems: [
      {
        text: 'header.headerCategory.columnOne.strollers',
        path: '/shop/new-strollers'
      },
      {
        text: 'header.headerCategory.columnOne.quatro',
        path: '#'
      },
      {
        text: 'header.headerCategory.columnOne.lorelli',
        path: '#'
      },
      {
        text: 'header.headerCategory.columnOne.carello',
        path: '#'
      },
    ]
  },
  {
    id: 'Модульные',
    listItems: [
      {
        text: 'header.headerCategory.columnTwo.modules',
        path: '/shop/new-prams'
      },
      {
        text: 'header.headerCategory.columnTwo.adamex',
        path: '#'
      },
      {
        text: 'header.headerCategory.columnTwo.used',
        path: '#'
      }
    ]
  },
  {
    id: 'Автолюльки',
    listItems: [
      {
        text: 'header.headerCategory.columnThree.carSeats',
        path: '/shop/new-car-seats'
      },
      {
        text: 'header.headerCategory.columnThree.carlo',
        path: '#'
      },
      {
        text: 'header.headerCategory.columnThree.adamex',
        path: '#'
      },
      {
        text: 'header.headerCategory.columnThree.used',
        path: '#'
      }
    ]
  },
  {
    id: 'Аксесуары',
    listItems: [
      {
        text: 'header.headerCategory.columnFour.accessories',
        path: '/shop/accessories'
      },
      {
        text: 'header.headerCategory.columnFour.adapters',
        path: '/shop/accessories'
      },
      {
        text: 'header.headerCategory.columnFour.tires',
        path: '/shop/accessories'
      },
      {
        text: 'header.headerCategory.columnFour.cameras',
        path: '/shop/accessories'
      }
    ],
  },
  {
    id: 'Постельное белье',
    listItems: [
      {
        text: 'header.headerCategory.columnFive.linens',
        path: '/shop/linens'
      },
      {
        text: 'header.headerCategory.columnFive.individualTailoring',
        path: '#'
      }
    ]
  }
];


export const headerArticlesData = [
  {
    text: 'header.headerArticle.articleOne.title',
    path: '#'
  },
  {
    text: 'header.headerArticle.articleTwo.title',
    path: '#'
  },
  {
    text: 'header.headerArticle.articleThree.title',
    path: '#'
  }
];

