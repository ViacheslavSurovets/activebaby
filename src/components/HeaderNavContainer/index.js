export { default as HeaderNavContainer } from './HeaderNavContainer';
