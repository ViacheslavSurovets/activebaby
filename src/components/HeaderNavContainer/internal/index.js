export { HeaderCategory } from './HeaderCategory';
export { HeaderArticles } from './HeaderArticles';
