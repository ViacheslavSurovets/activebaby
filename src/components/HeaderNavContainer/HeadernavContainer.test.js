import React from 'react';
import { shallow } from 'enzyme';
import HeaderNavContainer from './HeaderNavContainer';


describe ( 'HeaderNavContainer', () => {
  it ( 'HeaderNavContainer Snapshot', () => {
    expect ( shallow ( <HeaderNavContainer /> ) ).toMatchSnapshot ();
  } );
} );
