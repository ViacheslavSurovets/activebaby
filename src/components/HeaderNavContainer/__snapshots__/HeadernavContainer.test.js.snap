// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNavContainer HeaderNavContainer Snapshot 1`] = `ShallowWrapper {}`;
